src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+
+
+public class Duke {
+    private final Ui ui;
+    private TaskList taskList;
+    private final Storage storage;
+
+
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage();
+        taskList = new TaskList();
+        try {
+            taskList = new TaskList(storage.load(filePath));
+        } catch (DukeException e) {
+            ui.printError(e.getMessage());
+            taskList = new TaskList();
+        }
+
+    }
+
+    public Duke() {
+        this(Storage.SERIALIZATION_PATH);
+    }
+
+
+    public void run() {
+        Ui.initialize();
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        String[] commandAndParameter = Parser.inputParser(input);
+        Command currentCommand = Parser.commandParser(commandAndParameter[0]);
+        String currentParameter = commandAndParameter[1];
+
+        do {
+            try {
+                System.out.println(currentCommand);
+                StringBuilder sb = new StringBuilder();
+                String[] descriptionAndTime;
```

Comment:
```suggestion
                String[] descriptionAndTimeParts;
```
According to the java coding standard, it may be clearer if you use plural form to represent a collection of objects, like an array.